var/ramble/repos/builtin/package_managers/environment-modules/package_manager.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2022-2024 The Ramble Authors
+#
+# Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+# https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+# <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+# option. This file may not be copied, modified, or distributed
+# except according to those terms.
+
+from ramble.pkgmankit import *  # noqa: F403
+
+import os
+import llnl.util.filesystem as fs
+
+import ramble.util.hashing
+
+
+class EnvironmentModules(PackageManagerBase):
+    """Definition for using environment-modules as a package manager
+
+    This definition allows experiments to use environment-modules to manage the
+    software used in an experiment. It assumes the `module` command will be in
+    the path of the experiment at execution time.
+    """
+
+    name = "environment-modules"
+
+    maintainers("douglasjacobsen")
+
+    def get_spec_str(self, pkg, all_pkgs, compiler):
+        return pkg.spec
+
+    register_phase(
+        "write_module_commands",
+        pipeline="setup",
+        run_before=["make_experiments"],
+    )
+
+    def populate_inventory(
+        self, workspace, force_compute=False, require_exist=False
+    ):
+        env_path = self.app_inst.expander.env_path
+
+        self.app_inst.hash_inventory["package_manager"].append(
+            {
+                "name": self.name,
+            }
+        )
+
+        env_hash = ramble.util.hashing.hash_file(
```

Comment:
I wonder if going forward we should abstract this to a interface where we just supply the files to hash?

Reply:
How would you want these hashes to be combined? (i.e. in the inventory we write a singular `digest` attribute, and I'd prefer that remained the same, so we'd need some nice way of combining the hashes of multiple files into a single digest)